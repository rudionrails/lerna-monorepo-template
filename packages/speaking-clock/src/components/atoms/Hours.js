import React from "react";
import PropTypes from "prop-types";

const Hours = ({ hours, minutes }) => (
  <div className="font-weight-bold">
    {minutes === 0 ? hours : `${hours} hours`}
  </div>
);

Hours.propTypes = {
  hours: PropTypes.number.isRequired,
  minutes: PropTypes.number.isRequired,
};

export default Hours;
