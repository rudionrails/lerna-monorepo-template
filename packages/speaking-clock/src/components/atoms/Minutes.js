import React from "react";
import PropTypes from "prop-types";

const Minutes = ({ minutes }) => (
  <div className="font-weight-light text-muted">
    {minutes === 0 ? "o' clock" : `and ${minutes} minutes`}
  </div>
);

Minutes.propTypes = {
  minutes: PropTypes.number.isRequired,
};

export default Minutes;
