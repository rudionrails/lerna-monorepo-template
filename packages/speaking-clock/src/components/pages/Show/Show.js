import React from "react";
import PropTypes from "prop-types";

import Hours from "../../atoms/Hours";
import Minutes from "../../atoms/Minutes";

const Show = ({ date }) => {
  const hours = date.getHours();
  const minutes = date.getMinutes();

  return (
    <div className="text-center">
      <div className="small text-muted">the time is</div>

      <Hours {...{ hours, minutes }} />
      <Minutes {...{ minutes }} />
    </div>
  );
};

Show.propTypes = {
  date: PropTypes.shape({
    getHours: PropTypes.func.isRequired,
    getMinutes: PropTypes.func.isRequired,
  }).isRequired,
};

export default Show;
