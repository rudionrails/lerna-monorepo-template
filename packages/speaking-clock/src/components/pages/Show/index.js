import { connect } from "../../../store";
import Show from "./Show";

export default connect(state => ({
  date: state.date,
}))(Show);
