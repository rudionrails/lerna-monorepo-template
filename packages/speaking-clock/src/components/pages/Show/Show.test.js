import React from "react";

import Hours from "../../atoms/Hours";
import Minutes from "../../atoms/Minutes";

// module under test
import Show from "./Show";

const hours = 10;
const minutes = 13;
const date = {
  getHours: () => hours,
  getMinutes: () => minutes,
};

let subject;

beforeEach(() => {
  subject = shallow(<Show {...{ date }} />);
});

test("to render Hours", () => {
  const h = subject.find(Hours);

  expect(h.exists()).toBe(true);
  expect(h.props()).toEqual({ hours, minutes });
});

test("to render Minutes", () => {
  const m = subject.find(Minutes);

  expect(m.exists()).toBe(true);
  expect(m.props()).toEqual({ minutes });
});

test("to match snapshot", () => {
  expect(subject).toMatchSnapshot();
});
