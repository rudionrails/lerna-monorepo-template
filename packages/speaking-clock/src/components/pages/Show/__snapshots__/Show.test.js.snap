// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snapshot 1`] = `
<div
  className="text-center"
>
  <div
    className="small text-muted"
  >
    the time is
  </div>
  <Hours
    hours={10}
    minutes={13}
  />
  <Minutes
    minutes={13}
  />
</div>
`;
